synapse/handlers/pagination.py

```diff
@@ -335,20 +335,18 @@ async def get_messages(
         user_id = requester.user.to_string()
 
         if pagin_config.from_token:
-            room_token = pagin_config.from_token.room_key
+            from_token = pagin_config.from_token
         else:
-            pagin_config.from_token = (
-                self.hs.get_event_sources().get_current_token_for_pagination()
-            )
-            room_token = pagin_config.from_token.room_key
+            from_token = self.hs.get_event_sources().get_current_token_for_pagination()
 
-        room_token = RoomStreamToken.parse(room_token)
+        if pagin_config.limit is None:
+            # This shouldn't happen as we've set a default limit before this
+            # gets called.
+            raise Exception("limit not set")
```

Comment:
I think we've been usually use `assert` for checks like this?

Reply:
My reasoning here was that nothing stops someone from calling this from somewhere else, so it feels more plausible that this could happen